pandas/core/indexes/category.py

```diff
@@ -130,6 +130,9 @@ def _create_categorical(self, data, categories=None, ordered=None):
         -------
         Categorical
         """
+        if isinstance(data, ABCSeries) and is_categorical_dtype(data):
```

Comment:
The `isinstance(data, ABCSeries)` is maybe a bit too restrictive. This rules out, e.g. CategoricalIndex. But at moment the CategoricalIndex constructor takes a different (faster) path, so it wouldn't benefit from this change anyway.

Reply:
hmm, does it still pass if you add ``ABCCategoricalIndex`` here as well? it is more consistent

Reply:
How about `isinstance(data, (ABCSeries, type(self)))`. That'll catch `Series(dtype='category')`,  and `CategoricalIndex`,  but not `Categorical`.

Reply:
how about (is_categorical_dtype and not ABCCategorical)?

*maybe* add a helper method

``_ensure_categorical`` ?

which passes thru Categorical
does .values on CI and Series
and raises else

Reply:
> maybe add a helper method

I'd prefer to wait till we need that. I haven't come across any other cases where I need to treat CategoricalIndex and Series(dtype=category) differently than Categorical.